test/utils/describeConformance.js

```diff
@@ -79,10 +79,20 @@ function testClassName(element, getOptions) {
  */
 function testComponentProp(element, getOptions) {
   describe('prop: component', () => {
-    it('can render another root component with the `component` prop', () => {
-      const { classes, mount, testComponentPropWith: component = 'em' } = getOptions();
-
-      const wrapper = mount(React.cloneElement(element, { component }));
+    it('can render another root component with the `component`/`components` prop', () => {
+      const {
+        classes,
+        mount,
+        testComponentPropWith: component = 'em',
+        componentsProp,
+      } = getOptions();
+
+      const wrapper = mount(
+        React.cloneElement(
+          element,
+          componentsProp ? { components: { Root: component } } : { component },
+        ),
+      );
```

Comment:
Why do we abstract it instead of forking the conformance suite? These are two different sets of components so they should have two different suites. Please abstract this later.

Reply:
Let me create a separate set of conformance suite for these components in a new PR.

Reply:
Done in https://github.com/mui-org/material-ui/pull/23798